### PR TITLE
Add GitHub Actions

### LINKED ISSUE #64: Migrate CI to GitHub Actions
Travis CI is dead, long live GitHub Actions!

### PR DESCRIPTION
**Pull request by bollwyvl**
Hello again, and thanks for genanki!

Here's the smallest possible github actions workflow I could put together. Many other bells and whistles could be added, but this goes pretty far to getting more platform/python coverage.

For a look at what this is like, here's a [self-PR](https://github.com/bollwyvl/genanki/pull/1) and a [run](https://github.com/bollwyvl/genanki/actions/runs/663127404).

References:
- fixes #64

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+defaults:
+  run:
+    shell: bash -l {0}
```
Comment (kerrickstaley) [inline .github/workflows/ci.yml:13]: Could you please add a comment describing the purpose of setting the default shell here? Do the tests work if you simply remove this whole `defaults` block?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  ANKI_TEST_REVISION: ed8340a4e3a2006d6285d7adf9b136c735ba2085
+
+jobs:
+  test:
+    runs-on: ${{ matrix.os }}-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu, macos, windows]
+        py: [3.6, 3.8, 3.9]
```
Comment (kerrickstaley) [inline .github/workflows/ci.yml:25]: Please name this `python-version` for clarity.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        run: sudo apt-get install portaudio19-dev python3-dev
+      - if: ${{ matrix.os == 'macos' }}
+        run: brew install portaudio
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py }}
+      - run: ${{ matrix.py-cmd }} -m pip install --upgrade pip wheel setuptools
+      - if: ${{ matrix.os == 'windows' }}
+        run: ${{ matrix.py-cmd }} -m pip install pywin32
+      - run: ${{ matrix.py-cmd }} -m pip install -e ".[test]" pytest-cov codecov
```
Comment (kerrickstaley) [inline .github/workflows/ci.yml:51]: Could you add a comment like

    # The ".[test]" syntax installs the packages in install_requires and tests_require from setup.py. See https://stackoverflow.com/a/43893575

**Review by kerrickstaley — CHANGES_REQUESTED** [review]
Thank you so much for the contribution! Glad to be able to scratch this off my to-do list :smile:  Just requesting a few small changes.

**Comment by bollwyvl** [comment]
Ran clean with suggested changes applied:
https://github.com/bollwyvl/genanki/pull/1/checks?check_run_id=2210504062

File: .github/workflows/ci.yml (hunk 3 above)
Comment (bollwyvl) [reply .github/workflows/ci.yml]: added the comment, but the full URL, etc. seems a bit overkill... if anything, i'd recommend normalizing the `test_require`, which is deprecated, to use the `extras_require` syntax.

**Review by kerrickstaley — APPROVED** [review]

**Comment by kerrickstaley** [comment]
Looks great, thanks!

**Pull request merged**